test/e2e/ingress.go

```diff
@@ -135,6 +143,32 @@ var _ = framework.KubeDescribe("Loadbalancing: L7 [Feature:Ingress]", func() {
 			By("should reject HTTP traffic")
 			ExpectNoError(pollURL(fmt.Sprintf("http://%v/", ip), "", lbPollTimeout, httpClient, true))
 
+			By("should have correct firewall rule for ingress")
+			fw, err := getFwForIngress(cloudConfig)
+			Expect(err).NotTo(HaveOccurred())
+			// Retrieve nodeTags from the node instance as the target tags for the ingress firewall rule
+			nodes := framework.GetReadySchedulableNodesOrDie(jig.client)
+			Expect(len(nodes.Items) > 0).Should(BeTrue())
+			nodeTags, err := getInstanceTags(cloudConfig, nodes.Items[0].Name)
+			Expect(err).NotTo(HaveOccurred())
+			// Getting ingress backend nodePorts. Current GCE ingress controller allows
+			// traffic to the default backend nodePort by default, so retrieve that as well.
+			nodePorts := []string{}
+			defaultSvc, err := jig.client.Core().Services(api.NamespaceSystem).Get(defaultBackendName)
+			Expect(err).NotTo(HaveOccurred())
+			nodePorts = append(nodePorts, strconv.Itoa(int(defaultSvc.Spec.Ports[0].NodePort)))
+			// Test service need to be consistent with the content in below file:
+			// test/e2e/testing-manifests/ingress/static-ip.svc.yaml
+			testSvc, err := jig.client.Core().Services(ns).Get("echoheaders-https")
+			Expect(err).NotTo(HaveOccurred())
+			nodePorts = append(nodePorts, strconv.Itoa(int(testSvc.Spec.Ports[0].NodePort)))
+			expFw, err := constructFwForIngress(fw.Name, nodePorts, nodeTags.Items)
+			Expect(err).NotTo(HaveOccurred())
+			// Passed the last argument as true to verify the backend ports are
+			// included but not exactly matched, given there may be other existing
+			// ingress resources and backends we are not aware of.
+			Expect(verifyFirewallRule(fw, expFw, cloudConfig.Network, true)).NotTo(HaveOccurred())
```

Comment:
I think you're missing a file in your commit? since I can't seem to find some methods.
Suggest having this as a method on the gceController like
```
func (g *gceController) verifyFirewall(ingName string, expectedTags []string{}, expectedPorts []int{}) {

    // recontruct the firewall name from ing name like https://github.com/kubernetes/kubernetes/blob/master/test/e2e/ingress_utils.go#L601
    // get the firewall object
    // insert all its ports into a sets.String
    // insert the array of given ports into a sets.String
    // check IsSuperSet
    // verify tags

}
```